### DISCUSSION TITLE
Better tool to define the exact center freqency of FSK signals

### DISCUSSION THREAD

**Discussion by Bigblock** [jopohl/urh#863] [Ideas]
I often have the problem that I have to determine the exact center of the two signal peaks of FSK signals in URH.

Would be nice to have the possibilty to set the center of the two signal peaks in the graph and than URH would automatically calculate the center between them; so you have the exact center frequency.

I'm not a programmer. So i can only offer the idea.

![fsk-center-frequency](https://user-images.githubusercontent.com/1734860/116658978-2a4d6780-a991-11eb-8d0d-a307e77e1176.png)

**Comment by jopohl** [comment]
Basically, this is what the backend does when you click the "Autodetect parameters" button.

**Reply by Bigblock** [reply]
Yes, i understand that. But sometimes when signals are not 100% clean (or maybe there is a phase problem, or my capture hardware is not good enough - not shure what causes the problem) the automatic function seams not to work perfect. In that case i always will make some more measurements, but they all bring wrong results. 

I use URH to check if the Manchester encoded FSK signal is decoded correctly and if the transmitter works as it should. If the center of the FSK signal is not found correct by the automatic parameters button i get a wrong result. So when this happens the transmitter will be accepted by the receiver, but the signal decoded by URH is off, which to me would mean the transmitter doesn't work as it should.

So i would prefer to have a more precise manual way / tool to mark the 2 peaks of the FSK signal, to make shure that the center is found correctly.

PS: sorry for my bad english.

**Reply by jopohl** [reply]
Doesn't setting the center in the demodulated view solve this? You can move the center in the graph with the mouse and the rectangular signal makes it easy to separate the two different frequencies.

**Reply by Bigblock** [reply]
It depends. If the frequency graph is clean like in my first post, most times it's OK. If i have to check the whole signal it's very time consuming, because some sequences are very long.

There are also cases where the chart is not that clean, but the transmitter works perfectly. So i would pefer to have a more manual way to define the 2 peaks in an FSK signal.

An example:
![example](https://user-images.githubusercontent.com/1734860/117000858-f9976600-ace1-11eb-91b7-4d6340cb9ef3.jpg)

In general i use auto detection and it works very good. But sometimes i get strange results where i don't know what causes them. So i would prefer to have a little more manual control when defining the 2 peaks in the FSK signal.

PS: sorry for my late response.
